cmd/serverless/main.go

```diff
@@ -116,6 +117,12 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 
 	outputDatadogEnvVariablesForDebugging()
 
+	startChecker := startchecker.InitStartChecker()
```

Comment:
nit - I like the abstraction, but it feels a bit premature as we've added 150 LOC to check presence of 3 env vars and to return an error if none are present.

Is there another checker we think we'll add soon that makes this investment worthwhile now?

Reply:
I think I agree with AJ here. This could reasonably be three if statements and a single function.

Reply:
alright! I was designing this to be `open for extension closed for modification` but since you both agree that's definitely overkill, I'll make it simpler :)